aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -680,6 +680,19 @@ Tensor _unsafe_masked_index(const Tensor& self, const Tensor& mask, const torch:
     return at::clamp(*index, -size, size - 1);
   };
 
+  if (self.numel() == 0) {
+    std::vector<int64_t> new_size(self.dim());
+    auto get_size = [](const c10::optional<Tensor>&index, auto size) -> int64_t {
+      if (!index) {
+        return size;
+      } else {
+        return index->numel();
```

Comment:
I think you're assuming the index tensor here is a 1d-tensor viewed in multiple dimensions, but I don't think that's always the case.

Reply:
I didn't think we would or need to support anything other than 1d-tensors as elements of the indices list. I don't know what a higher dimensional tensor here is supposed to do.

Reply:
Here is an example of multidimensional indexing:
```python
import torch
a = torch.arange(100).reshape(10, 10)
b = torch.randn(100)
b[a] .shape # (10, 10)
```

Reply:
A quick and dirty way to get the right output shape might be to call the meta implantation for `at::index`